app/controllers/idv/review_controller.rb

```diff
@@ -124,6 +125,13 @@ def password
       params.fetch(:user, {})[:password].presence
     end
 
+    def confirm_verify_info_complete
```

Comment:
I'm looking at `#confirm_idv_steps_complete` which is also a before action and has a call to `#idv_profile_complete?`. Wdyt of wrapping this logic into that method? That way a user can confirm in either the FSM or the new controller and still pass the before actions here.

Reply:
Sadly it doesn't to that before action, but makes a redirect loop between review and verify_info somewhere before that. I couldn't figure out where, but putting this before action right at the top fixed it.

Reply:
Paired on this with the team. We figured out that the problem is a little complicated. The process looks like this:

1. The user fails proofing at the verify info controller
2. The user is redirected to the idv session error page
3. There is a link on the idv session error page that redirects back to the doc auth FSM
4. The user enters the doc auth FSM which sees that all steps are complete and redirects to the review controller
5. The review controller runs the `confirm_idv_session_started` before action which checks for an applicant. Since resolution did not succeed there is no applicant. This causes it to redirect back to the doc auth FSM
6. The doc auth FSM proceeds as if in step 4 and begins thrashing

With all that in mind we are all convinced the before action is probably the best route. We will have to figure out how to handle the redirects back to the flow state machine or our new code when we handle the deployment ticket (LG-7402).